### PR TITLE
Alex/sat 1580

### PR DESCRIPTION
**Pull request by choroshin**

### CODE REVIEW

File: src/components/diagnosticExercise/services/workoutsDiagnosticFlow.service.js [hunk 1]
```diff
@@ -211,7 +212,87 @@
...
+                workoutsDiagnosticFlowObjApi.getQuestionsByDifficultyAndOrder = function (questions, difficulty, order, cb) {
+                    $log.debug('WorkoutsDiagnosticFlow getQuestionsByDifficultyAndOrder: initial func', arguments);
+                    var diagnosticFlowResults = {};
+
+                    //in case initQuestionsByDifficultyAndOrder function was not called.
+                    if (!questionsByOrderAndDifficultyArr) {
+                        $log.debug('WorkoutsDiagnosticFlow getQuestionsByDifficultyAndOrder: questionsByOrderAndDifficultyArr is null, calling initQuestionsByDifficultyAndOrder function');
+                        workoutsDiagnosticFlowObjApi.initQuestionsByDifficultyAndOrder(questions);
+                    }
+                    if (angular.isArray(questionsByOrderAndDifficultyArr) && questionsByOrderAndDifficultyArr.length === 0) {
```
Comment (Ariel-W) [inline src/components/diagnosticExercise/services/workoutsDiagnosticFlow.service.js]: add if (!questionsByOrderAndDifficultyArr) || ...

File: src/components/diagnosticExercise/controllers/workoutsDiagnosticExercise.controller.js [hunk 2]
```diff
@@ -212,16 +212,17 @@
...
 
             ZnkExerciseUtilitySrv.setQuestionsGroupData(exerciseData.questionsData.questions, exerciseData.questionsData.questionsGroupData, exerciseData.resultsData.playedAudioArticles);
 
             // init question and questionResults for znk-exercise
             if (!diagnosticSettings.isFixed) {
+                WorkoutsDiagnosticFlow.initQuestionsByDifficultyAndOrder(exerciseData.questionsData.questions);
+
                 if (resultsData.questionResults.length === 0) {
-                    WorkoutsDiagnosticFlow.getQuestionsByDifficultyAndOrder(questions, resultsData.questionResults, mediumLevelNum, numQuestionCounter + 1, function (diagnosticFlowResults) {
+                    WorkoutsDiagnosticFlow.getQuestionsByDifficultyAndOrder(exerciseData.questionsData.questions, mediumLevelNum, numQuestionCounter + 1, function (diagnosticFlowResults) {
```
Comment (Ariel-W) [inline src/components/diagnosticExercise/controllers/workoutsDiagnosticExercise.controller.js]: use "questions"

**Comment by assafshp** [comment]
fine with me

File: src/components/diagnosticExercise/controllers/workoutsDiagnosticExercise.controller.js (hunk 2 above)
Comment (choroshin) [inline src/components/diagnosticExercise/controllers/workoutsDiagnosticExercise.controller.js]: done

**Pull request merged**